### ISSUE TITLE
AbbreviationAsWordInName - Unit tests - Methods with generic arguments

### ISSUE THREAD

**Issue by ZabuzaW** [checkstyle/checkstyle#5787]
Suppose we have a method like
```java
public foo(N node, E edge, double bar) {
  ...
}
```
IDEs like Eclipse and Intellij will create the following JUnit test for this method:
```java
@Test
public void testFooNEDouble() {
  ...
}
```
The module `AbbreviationAsWordInName` does **not account** for this scenario, nor does it support functionality to skip a method name like this (for example with a prefix of _test_).

The module reads the `NED` and will **warn** if the property `allowedAbbreviationLength` is set to a value less than `3`, like with Googles configuration:
```xml
<module name="AbbreviationAsWordInName">
  <property name="ignoreFinal" value="false"/>
  <property name="allowedAbbreviationLength" value="1"/>
</module>
```
The obvious problem is that a method could have an arbitrary amount of generic arguments.

The **alternative** would be to declare a method name like this as not preferred and to propose an alternative method name.

**Comment by romani** [comment]
when smth is generating a code and have to blindly follow it, style rules works badly.  You have to use suppressions to avoid violations or even disable checkstyle at all on generated code. Generated code is not suppose to be read by human so styling it does not make sense.

In my experience, generations from IDE are default template for engineer to start coding. BUT it is never considered as not his fault. As engineer committed such code he get full ownership of changed code, IDEs default proposals considered as code written by human, as there is no guaranty you was forced to make such method.

if you do HAVE to follow that weird code generation - use suppressions, or separate config for test code or any other workaround.
We will not update Check to support such cases. `testFooNEDouble` is terrible name and there is no excuse to have it in code in my objective opinion.
